VM_SETUP.md

```diff
@@ -0,0 +1,179 @@
+        packages:
+        - git
+        - build-essential
+        - jq
+        - snapcraft
+        runcmd:
+        - cat /dev/zero | ssh-keygen -q -N ""
+        ssh_authorized_keys:
+        - $(cat ${HOME}/.ssh/id_rsa.pub | cut -d' ' -f1-2)
+description: MAAS Anvil environment
+devices:
+    root:
+        path: /
+        pool: default
+        type: disk
+        size: 50GB
+    work:
+        type: disk
+        source: ${MAAS_ANVIL_SRC}
+        path: /work
+    eth0:
+        type: nic
+        name: eth0
+        network: maas-anvil-ctrl
+    eth1:
+        type: nic
+        name: eth1
+        network: maas-anvil-kvm
+EOF
+```
+
+## Launch the VMs
+
+To create your VMs, execute the following:
+
+```bash
+lxc launch ubuntu:22.04 infra1 --vm -p ${PROFILE_NAME}
+lxc launch ubuntu:22.04 infra2 --vm -p ${PROFILE_NAME}
+lxc launch ubuntu:22.04 infra3 --vm -p ${PROFILE_NAME}
+```
+
+If you are running these commands in a script, include the lines below to wait for the VMs to be ready.
+
+```bash
+# this sleep may not be necessary, or may be shortened depending on your system
+sleep 10
+lxc exec infra1 -- cloud-init status --wait
+lxc exec infra2 -- cloud-init status --wait
+lxc exec infra3 -- cloud-init status --wait
+```
+
+## Set up VMs Internally and Install MAAS Anvil
+
+We now need to execute some commands in each VM. This will set up our second network interface, setup LXD inside the VM, and build & install the MAAS Anvil snap.
+
+```bash
+KVM_NETWORK_PREFIX="10.40.0"
+
+sudo tee /etc/netplan/99-maas-kvm-net.yaml <<EOF
+network:
+    version: 2
+    ethernets:
+        enp6s0:
+            addresses:
+                - ${KVM_NETWORK_PREFIX}.2/24
+EOF
+sudo netplan apply
+
+lxd init --auto
+
+sleep 5
+cd /work
+/usr/bin/make snap
+sudo snap install ./maas-anvil.snap --dangerous
+```
+
+To execute this inside the VM from a script running on your host machine, save the contents above to a file (below, `/path/to/script`), then run:
+
+```bash
+infra1_ip=$(lxc list -c4 --format csv infra1 | grep -o "10.30.0.[0-9][0-9]")
+ssh -o "StrictHostKeyChecking no" ubuntu@${infra1_ip} bash -s < /path/to/script
+```
+
+## Set up MAAS Anvil snap
+
+Finally, we need to:
+
+- Delete the default bridge interface created by `lxd init --auto`, as having this interface active may cause the MAAS region charm to assign itself to this network.
+- Run the `prepare-node-script` provided by MAAS Anvil
+- Login to the `snap_daemon` group
+- Create a self-signed SSL certificate, to optionally be used for MAAS Anvil TLS configuration
+
+```bash
+sudo ip link delete lxdbr0
+
+maas-anvil prepare-node-script | bash -x
+
+newgrp snap_daemon
+
+openssl req -new -x509 -nodes -subj "/C=US/ST=Colorado/L=Denver/O=MAAS/OU=Anvil/CN=infra1.maas-anvil-ctrl" -keyout /home/ubuntu/.config/anvil/key.pem -out /home/ubuntu/.config/anvil/cert.pem -days 3650
+```
+
+You should now have three VMs setup: `infra1`, `infra2`, and `infra3` where you can bootstrap/join the cluster.
```

Comment:
Should we add a section at the bottom for installing changes to the snap?
(I mean, I know it's just make the snap and install again, but might be useful)

Reply:
So the only issue I have with this process here is that by deleting the `lxdbr0` interface, you can't build the snap again without resetting the entire VM (at least, as far as I know). You can't just bring down the interface either, as the maas-region charm will still try to tell HA Proxy that's where it lives. If you have other suggestions, let me know.

Reply:
Hmm, yes that's fair. Given that spinning up a VM is so quick, there's not really any use to reinstalling a new version vs. jsut spinning up the fresh build.

Reply:
OK, I added a note about deleting the VMs and starting over